packages/block-library/src/image/edit.native.js

```diff
@@ -261,6 +294,56 @@ class ImageEdit extends React.Component {
 			</BottomSheet>
 		);
 
+		//Used for display of Inspector Controls
+		const getSizeSlugDisplay = ( sizeSlugValue ) => {
+			if ( sizeSlugValue === undefined ) {
+				sizeSlugValue = DEFAULT_SIZE_SLUG;
+			}
+			const sizeDisplayItems = getSizeOptionsItems();
+			switch ( sizeSlugValue ) {
+				case IMAGE_SIZE_THUMBNAIL:
+					return sizeDisplayItems[ 0 ].label;
+				case IMAGE_SIZE_MEDIUM:
+					return sizeDisplayItems[ 1 ].label;
+				case IMAGE_SIZE_FULL_SIZE:
+					return sizeDisplayItems[ 3 ].label;
+				case IMAGE_SIZE_LARGE:
+				default:
+					return sizeDisplayItems[ 2 ].label;
+			}
+		};
+
+		const getSizeOptionsItems = () => {
+			return [
+				{ value: IMAGE_SIZE_THUMBNAIL, label: __( 'Thumbnail' ) },
+				{ value: IMAGE_SIZE_MEDIUM, label: __( 'Medium' ) },
+				{ value: IMAGE_SIZE_LARGE, label: __( 'Large' ) },
+				{ value: IMAGE_SIZE_FULL_SIZE, label: __( 'Full Size' ) },
+			];
+		};
+
+		let picker;
+
+		const getSizeOptions = () => (
+			<Picker
+				hideCancelButton={ true }
+				title={ __( 'Image Size' ) }
+				ref={ ( instance ) => picker = instance }
+				options={ getSizeOptionsItems() }
+				leftAlign={ true }
+				onChange={ ( value ) => {
+					this.onSetSizeSlug( value );
+				} }
+			/>
+		);
+
+		const onPickerPresent = () => {
```

Comment:
probably worth having it as a bound class method instead

Reply:
I moved this code to ImageOptionsPicker, but have not changed it to a bound class method yet. I'm using [this pattern from MediaUpload](https://github.com/WordPress/gutenberg/blob/a400f7b354a6f5dc3dd59daef3d4a90069dff54b/packages/block-editor/src/components/media-upload/index.native.js#L91-L103) for passing the presentPicker function to the container of our component. 
(github insert code kept deformating this code)
![image](https://user-images.githubusercontent.com/1103838/64218162-63100300-ce75-11e9-994e-579dd32509c9.png)

If we simply move onPickerPresent to be a class method, we'll no longer have a reference to picker that is assigned in the Picker props to the picker instance. 
The current code feels like an antipattern to me, but I figured consistency was a quick, safe option, and that maybe you or someone else on the team would have feedback on how we could update this code in MediaUpload and here at once. If the cleanest way to do this requires a little refactoring of the Picker, perhaps it is not the best time to do this though since the same pattern is used elsewhere in the code.

Reply:
I don't think this is a pattern we should try to reproduce here. It doesn't follow the SOC principle and makes the code harder to read imo. I agree MediaUpload is confusing in its current stage though.

Reply:
Utilizing `BottomSheet.PickerCell` allowed me to get rid of this code, but unfortunately, `PickerCell` is using the same pattern. I think all of this calling child methods from parent are stemming from `presentPicker()` method in the [ios ](https://github.com/WordPress/gutenberg/blob/8dc1de9ea27e43a3c8b32a6e23f5548ae9c552b3/packages/components/src/mobile/picker/index.ios.js#L13) and [android](https://github.com/WordPress/gutenberg/blob/8dc1de9ea27e43a3c8b32a6e23f5548ae9c552b3/packages/components/src/mobile/picker/index.android.js#L29) picker classes. I didn't have immediate ideas on how to refactor that, but have been doing some reading. Would be great to update so we stop breaking the SOC principle like you mentioned.